src/java.desktop/share/classes/javax/swing/plaf/basic/BasicSeparatorUI.java

```diff
@@ -152,5 +152,13 @@ public Dimension getPreferredSize( JComponent c )
     }
 
     public Dimension getMinimumSize( JComponent c ) { return null; }
-    public Dimension getMaximumSize( JComponent c ) { return null; }
+    public Dimension getMaximumSize( JComponent c ) {
+        Dimension d = getPreferredSize(c);
+        if (((JSeparator)c).getOrientation() == JSeparator.VERTICAL) {
+            return new Dimension(d.width, Integer.MAX_VALUE);
+        }
+        else {
+            return new Dimension(Integer.MAX_VALUE, d.height);
+        }
```

Comment:
`WindowsToolBarSeparatorUI` uses `Short.MAX_VALUE` instead of `Interger`:

https://github.com/openjdk/jdk/blob/ee3e0917b393b879a543060ace2537be84f20e82/src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsToolBarSeparatorUI.java#L74-L81

Should we follow the pattern here?

In Java Code Style, the `else` keyword should be on the same line as the closing brace.